x-pack/plugins/apm/public/components/app/settings/agent_explorer/agent_latest_version/index.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiSkeletonRectangle, EuiToolTip, useEuiTheme } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { NOT_AVAILABLE_LABEL } from '../../../../../../common/i18n';
+import { AgentName } from '../../../../../../typings/es_schemas/ui/fields/agent';
+
+export function AgentLatestVersion({
+  agentName,
+  isLoading,
+  latestVersion,
+  failed,
+}: {
+  agentName: AgentName;
+  isLoading: boolean;
+  latestVersion?: string;
+  failed?: boolean;
```

Comment:
is there any reason why these are optional?

Reply:
You could receive for example an undefined as lastVersion, imagine an agent we don't have covered (`OCaml` for instance). For the failed, good catch, shouldn't be ⭐

Reply:
`latestVersion?: string` part was addressed in [7b2cefb](https://github.com/elastic/kibana/pull/153643/commits/7b2cefb4943bce8e0da6cb80b547d1f3a0d48db4).